tron/kubernetes.py

```diff
@@ -338,12 +349,14 @@ def handle_next_event(self, _=None) -> None:
         * control: events regarding how the task_proc plugin is running - handled directly
         * task: events regarding how the actual tasks/Pods we're running our doing - forwarded to KubernetesTask
         """
-        if self.deferred and not self.deferred.called:
+        if self.deferred is not None and not self.deferred.called:
             log.warning("Already have handlers waiting for next event in queue, not adding more")
             return
 
         self.deferred = self.queue.get()
-
+        if self.deferred is None:
+            log.warning("Unable to get a handler for next event in queue - this should never happen!")
+            return
```

Comment:
hmm, i wonder if we should crash here - if this has happened we're going to completely stop processing events

Reply:
or maybe we should add in a fake deferred so that we can still add in the callbacks and retry later?

Reply:
we tend to return if this happens with the log warning message

Reply:
if that happens, we'll never process events again since we setup a psuedo loop here with the callbacks - which is not great since we'll likely not notice this until it's too late to easily recover the current state of things with a tron restart given that the completed pods will have aged out by the time we notice this